### PR TITLE
highlight: Move client into internal package

### PR DESCRIPTION
**Pull request by efritz**
Pulled from #51690.

## Test plan

Existing CI.

### CODE REVIEW

File: internal/highlight/highlight.go [hunk 1]
```diff
@@ -24,21 +24,19 @@ import (
...
 func LoadConfig() {
-	syntectServer := env.Get("SRC_SYNTECT_SERVER", "http://syntect-server:9238", "syntect_server HTTP(s) address")
-	client = gosyntect.New(syntectServer)
+	client = gosyntect.GetSyntectClient()
 }
 
-var client *gosyntect.Client
-
 var (
+	clientOnce      sync.Once
```
Comment (Strum355) [inline internal/highlight/highlight.go:38]: Is this used anywhere? Cant do find-refs coz no ones fixing the extension..

Comment (efritz) [reply internal/highlight/highlight.go:38]:
```suggestion
```

Not yet actually

**Comment by Strum355** [comment]
No objections to moving, but just a quick question on whether theres non-frontend areas that will be consuming this? A quick look at Ye Ol' Bazel Graph shows its currently only consumed by {enterprise,}/cmd/frontend packages

**Comment by efritz** [comment]
> No objections to moving, but just a quick question on whether theres non-frontend areas that will be consuming this? A quick look at Ye Ol' Bazel Graph shows its currently only consumed by {enterprise,}/cmd/frontend packages

#51690 is about to use it, wanted to do the refactor things separately.

**Review by Strum355 — APPROVED** [review]

**Pull request merged**